### PR TITLE
cjdns: bump v21.1 and patch find-python

### PR DESCRIPTION
**Pull request by wfleurant**
**Maintainer**: me
**Compile tested**: ubuntu 22 (with Python 3.10 and 2.7) for rockchip-armv8
**Run tested**: rockchip-armv8 ... host and staging_dir has python3, we need python2 to compile gyp for libuv.

**Description**: lets bump to latest v21.1 protocol -- non breaking changes.  
This should fix compiling on systems with various versions of python.  
Patch wise, we try to use python2 and other python3 versions known to work.  
Python 3.10 requires a newer version of gyp to resolve collections.abc.mutableset changes.  
If users select `lang/python`, then cjdns will find and utilize 3.10 in staging before known-working versions on the host.
Long story short, Python 3.10 is not supported.  

Fixes issue reported in #882

### CODE REVIEW

**Comment by wfleurant** [comment]
i need some help what does this even mean...

```
Applying patch 001-five-mins-builder-zonk.patch
patching file node_build/builder.js

Applying patch 010-musl-12x.patch
patching file memory/Allocator.c
Hunk #1 succeeded at 57 with fuzz 2.

Applying patch 020-prefer-python2.patch
patching file node_build/FindPython.js
Now at patch 020-prefer-python2.patch
touch "/home/build/openwrt/build_dir/target-powerpc_8540_musl/cjdns-cjdns-v21.1/.quilt_checked"

Applying patch 001-five-mins-builder-zonk.patch
patching file node_build/builder.js

Now at patch 001-five-mins-builder-zonk.patch
Refreshed patch 001-five-mins-builder-zonk.patch

Applying patch 010-musl-12x.patch
patching file memory/Allocator.c
Hunk #1 succeeded at 57 with fuzz 2.

Now at patch 010-musl-12x.patch
Refreshed patch 010-musl-12x.patch

Applying patch 020-prefer-python2.patch
patching file node_build/FindPython.js

Now at patch 020-prefer-python2.patch
Patch 020-prefer-python2.patch is unchanged

mkdir -p ./patches
rm -f ./patches/* 2>/dev/null >/dev/null


Dirty patches detected, please refresh and review the diff

Error: Process completed with exit code 1.
```

**Comment by PolynomialDivision** [comment]
@wfleurant You can refresh the patchset by doing

    make package/cdjns/refresh

**Comment by PolynomialDivision** [comment]
Could you maybe squash some commits together? I know it sounds nitpicking but from my point of view each commit should be able to built alone. If you now put the refresh of patches into a separate commit, the ci fails for the commit before. Your commits does not look so heavy that they need a seperate commit (at least from what I see).

**Comment by wfleurant** [comment]
yeah, for sure... squash and merge into openwrt/routing, no?

**Comment by PolynomialDivision** [comment]
Maybe cherry-pick also to 21.02 and 22.03?

**Comment by zhangguanzhang** [comment]
> Maybe cherry-pick also to 21.02 and 22.03?

Agree

**Comment by wfleurant** [comment]
@PolynomialDivision i would like to port v21.1 + supernodes in https://github.com/openwrt/routing/pull/884   
is this trivial? could you help with this?

**Comment by PolynomialDivision** [comment]
Trivial:
- https://github.com/openwrt/routing/pull/885
- https://github.com/openwrt/routing/pull/886

**Comment by zhangguanzhang** [comment]
need sync to branch 18.06

**Comment by wfleurant** [comment]
@PolynomialDivision is backport possible? i don't think i have a device to test on @zhangguanzhang have you tested the package on device

**Comment by PolynomialDivision** [comment]
> need sync to branch 18.06

18.06 was EOL already in december 2020?
